### PR TITLE
Accept arrays as options for OptionsConfigs

### PR DESCRIPTION
**Pull request by manuelpuyol**
I've found myself using a lot of array => Hash conversions, so this could help a lot!
https://github.com/primer/view_components/pull/51

## Use case

It is quite normal to have a constant indicating which options are valid such as
https://github.com/primer/view_components/blob/main/app/components/primer/button_component.rb#L25-L26

Building stories using those constants instead of enumerating them all over again is quite helpful, since if we change the component, the story doesn't need to change.

### CODE REVIEW

File: spec/view_component/storybook/controls/options_config_spec.rb [hunk 1]
```diff
@@ -15,6 +15,19 @@
...
+
+      context "using array options" do
+        let(:options) { ["red", "blue","yellow"] }
+
+        it_behaves_like "a controls config" do
+          subject { described_class.new(type, component, param, options, value, name: name) }
+
+          let(:value) { "blue" }
+          let(:param_value) { "blue" }
+
```
Comment (jonspalmer) [inline spec/view_component/storybook/controls/options_config_spec.rb:27]: Can we consolidate these declarations given they don't change?

File: spec/view_component/storybook/controls/options_config_spec.rb [hunk 2]
```diff
@@ -15,6 +15,19 @@
 
         let(:csf_arg_type_control_overrides) { { options: options } }
       end
+
+      context "using array options" do
+        let(:options) { ["red", "blue","yellow"] }
```
Comment (jonspalmer) [inline spec/view_component/storybook/controls/options_config_spec.rb:20]:
```suggestion
        let(:options) { ["red", "blue", "yellow"] }
```

File: spec/dummy/test/components/stories/kitchen_sink_component_stories.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ class KitchenSinkComponentStories < ViewComponent::Storybook::Stories
       number_pets 2
       sports %w[football baseball]
       select(:favorite_food, { hot_dog: "Hot Dog", pizza: "Pizza", burgers: "Burgers", ice_cream: "Ice Cream" }, "Ice Cream")
-      radio(:mood, { happy: "Happy", sad: "Sad", angry: "Angry", content: "Content" }, "Happy")
+      radio(:mood, ["Happy", "Sad", "Angry", "Content"], "Happy")
```
Comment (jonspalmer) [inline spec/dummy/test/components/stories/kitchen_sink_component_stories.rb:13]: I don't think we need to change the kitchen sink example here. If we have unit tests of the specifics that is enough. Kitchen Sink should stay as the most complicated example of all options. If there is value in detailing this option to take an array lets create a separate story.

**Review by jonspalmer — COMMENTED** [review]
Like the addition. Could we clean up a few things and this will be good to go.

**Comment by jonspalmer** [comment]
Thinking about this more Storybook controls allow you to pass an array vs a hash as options. Isn't a better solution just to pass the options array through to Storybook vs constructing a hash from them?

**Comment by manuelpuyol** [comment]
> Thinking about this more Storybook controls allow you to pass an array vs a hash as options. Isn't a better solution just to pass the options array through to Storybook vs constructing a hash from them?

Interesting! I wasn't sure they accepted arrays.
let me take a look into it :)

**Comment by jonspalmer** [comment]
this commit should allow options as an array 1f4e78e6870e3f2b6f7ef279225a0367f870abf4

**Pull request closed without merging**